rcl/src/rcl/wait.c

```diff
@@ -614,6 +614,154 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_wait_multiple(rcl_wait_set_t wait_sets[], const size_t num_wait_sets, int64_t timeout)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(wait_sets, RCL_RET_INVALID_ARGUMENT);
+  size_t i;
+  size_t j;
```

Comment:
@jacobperron nit: I have a slight preference for declaring loop variables within the `for` statement. Won't block if you prefer to keep things C89-ish.

Reply:
I guess my head was in C89 mode. Since we're targeting C99, I'm happy to change it.